### DISCUSSION TITLE
Adjusting Vibration Amplitude

### DISCUSSION THREAD

**Discussion by Neurosmith24** [kriswilk/buzzah#11] [Q&A] [answered]
Hi @kriswilk & @PWPInnovator898
In Tass' papers he describes the tactor contact with the fingertip using 0.5mm precompression of the skin and peak to peak amplitude of 0.03 mm to 0.1 mm (ref below). Not having experienced how LRAs work, is this vibrational amplitude adjusted in Buzzah! via the AMPLITUDE_MAX and AMPLITUDE_MIN settings? In other words, if I design my tactor housing so that the tactor contact protrudes through the housing hole and contacts the skin with 0.05mm pre-compression, how do I adjust the max vertical amplitude to be within 0.03 to 0.1 mm? Thanks.

“Neurons in the cutaneous core of the human thalamic somatic sensory nucleus [Ventral caudal (Vc)] respond to vibratory stimuli (with static 0.5 mm indentation and 0.1 mm vibration amplitude) quite selectively with a pronounced phase entrainment [15]...
The indentation of the stimulation contact surface is constant, e.g. 0.5 mm, throughout the entire vCRS delivery. This can be realized by a permanent fixation of the vibratory stimulation device. The peak to peak amplitude is small, e.g. 0.1 mm or only 0.03 mm."

**Comment by PWPInnovator898** [comment]
Jeff; 
These are important specs but very hard to accurately measure without some lab gear, which I lack access to at this time.
So, we've had to rely on my own empirical comparison test of the Synergic gloves from an early trial participant.
i.e., I adjusted the AMPLITUDE_MAX value to "feel the same" vibration intensity as the trial glove settings.
The AMPLITUDE_MIN parameter was added by request to Kris to enable supporting "double randomization" per Tass.   This feature adds to the time randomization (jitter) feature in the code  Tass believes amplitude randomization may lead to longer lasting benefits for PwP.
Bottom line: if you have a way to more quantitatively measure displacement, please let  me know.
My current vibration settings: 

... 
ACTUATOR SPECIFICATIONS: These MUST be set to match the specifications of the
actuator that you are using.
"""

# Type of actuator: "LRA" or "ERM"
ACTUATOR_TYPE = "LRA"
# Maximum drive voltage, in V
ACTUATOR_VOLTAGE = 2.5 * math.sqrt(2)
# Driving frequency, in Hz
ACTUATOR_FREQUENCY = 225

# COORDINATED RESET PARAMETERS: These are based on the published research. You are
# free to change them to alter the behaviour of the device.

# Vibration amplitude range, in %
AMPLITUDE_MIN = 80
AMPLITUDE_MAX = 127
# Duration of each vibration and subsequent pause, each in seconds
TIME_ON = 0.100
TIME_OFF = 0.066666666
# Duration of the relaxation period between sets of vibrations, in seconds
TIME_RELAX = 4 * (TIME_ON + TIME_OFF)
# Jitter, in %
JITTER = 23.5
# Mirror left / right hand vibration patterns
MIRROR = False
# Duration of the therapy session, in minutes
TIME_SESSION = 60

**Reply by Neurosmith24** [reply]
Thanks for the info. Nice that you got to trial the Synergic gloves and have that as a reference. For the 0.5 pre-compression, are you just comparing yours based on Synergic for that too or have you just been able to define the CAD model and 3D print with enough precision to get that tight of a tolerance?

**Reply by kriswilk** [reply, marked as answer]
Getting a 0.5mm tolerance on a 3D-printed part should be no problem. Most printers can handle layer heights down to 0.05 - 0.1 mm. As long as the feature is being printed in the Z-direction, it should be well resolved.

**Reply by Neurosmith24** [reply]
@PWPInnovator898 in this comment you have the amplitude as 225. But you've mentioned using 235 in other comments. Is the 225 a typo or is that what you're using rather than 235. Thx!

**Reply by PWPInnovator898** [reply]
I'm using 235 Hz as the primary vibration frequency, as that's the setting that produces the best response sensitivity for me, based on trial and error only, with no theory behind it.   FWIW, I recently began enabling random frequencies between stimulation bursts to further randomize the stimulation patterns.  This is not prescribed in any recent papers I am aware of, but my thought follows the benefits of random stimulation being key to long lasting therapy benefits, such as from the amplitude and temporal (jitter) randomization Stanford refers to in recent studies.

**Comment by kriswilk** [comment]
There is no easy way to relate the firmware parameters to the physical vibration amplitude. I'm not sure if there is any easy way to measure it, either. One way might be to use a sensitive dial indicator (a machinist's tool). However, the frequency of the vibration would make seeing the excursions of the needle on the scale difficult.

That said, the parameters responsible for the overall amplitude of the vibration are ACTUATOR_VOLTAGE (maximum drive voltage) and AMPLITUDE_MAX.

If @PWPInnovator898 has had experience with genuine hardware, I'd say his choice of actuator and parameters probably reflect a good starting point.

**Reply by Neurosmith24** [reply]
Thanks!
